front/src/applications/operationalStudies/hooks/useAutoUpdateProjection.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect } from 'react';
+
+import { useSelector } from 'react-redux';
+
+import type { InfraWithState } from 'common/api/osrdEditoastApi';
+import type { TrainScheduleWithDetails } from 'modules/trainschedule/components/Timetable/types';
+import {
+  updateSelectedTrainId,
+  updateTrainIdUsedForProjection,
+} from 'reducers/osrdsimulation/actions';
+import { getSelectedTrainId, getTrainIdUsedForProjection } from 'reducers/osrdsimulation/selectors';
+import { useAppDispatch } from 'store';
+
+const useAutoUpdateProjection = (
+  infra: InfraWithState,
+  trainIds: number[],
+  trainScheduleSummaries: TrainScheduleWithDetails[]
+) => {
+  const dispatch = useAppDispatch();
+  const currentProjection = useSelector(getTrainIdUsedForProjection);
+  const selectedTrainId = useSelector(getSelectedTrainId);
+
+  useEffect(() => {
+    if (infra.state === 'CACHED' && trainIds.length > 0) {
```

Comment:
To reduce the condition nesting a bit ?

```suggestion
    if (infra.state !== 'CACHED' || trainIds.length === 0) return
```

Reply:
Done !